var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -718,6 +729,7 @@ def url_for_version(self, version):
     depends_on("eigen@3:", when="+machdyn")
     depends_on("py-cython", when="+mliap+python", type="build")
     depends_on("py-cython", when="+ml-iap+python", type="build")
+    depends_on("py-mdi", when="+mdi", type="build")
```

Comment:
Is it only a build dependency? I would guess you also want it at runtime for it to be available in Python.